telemetry/specs/INT.mdk

```diff
@@ -652,8 +652,32 @@ both INT header stacks carry information for respective layers and need not be
 considered interfering with each other.
 
 A field in Ethernet, IP, or TCP/UDP should indicate if the
-INT header exists after the TCP/UDP header. We propose two options.
-
+INT header exists after the TCP/UDP header. We propose three options.
+
+* UDP destination port field: a new UDP port number (INT_TBD) will be assigned
+by IANA to indicate the existence of INT after UDP. This option supports two
+use cases:
+  - The original packet already has UDP header either as user application
+    protocol or as part of another UDP-based encapsulation such as VXLAN,
+    GENEVE, RoCEv2. INT is inserted after the UDP header with the UDP
+    destination port number changed to INT_TBD.
+    The original destination port number
+    is carried in the shim header for the INT sink to restore back when it
+    removes the INT stack from the packet.
+  - A new UDP header for INT is inserted between IP and the existing L4 header.
+    The protocol field of IP header is set to 17 for UDP and the original
+    IP protocol value is carried in the INT shim header.
+    In the new UDP header, INT_TBD is used as the destination port number.
+    The source port number of the new UDP header must be either 1) copied from the
+    original L4 header, 2) computed as a hash of the existing L4 header
+    fields (such as source/destination ports in case of TCP/UDP), or 3)
+    computed as a hash of the original outer 5 tuple. The second option
+    is recommended to provide ECMP/LAG load-balancing logic with the same degree of
+    entropy of the original data packet.
+The INT shim header has a field NPT (Next Protocol Type) that indicates
+which of the two cases are applied to a given INT packet. In case a new UDP
+header was inserted, the INT sink will strip the UDP header off from the
+packet together with the rest of INT headers.
```

Comment:
We should explicitly state the complete behavior that is expected from the INT sink.
For the case that original packet already had UDP header, INT sink must copy original destination port number into UDP header field and strip the INT header.
For the case that UDP header was added to the packet, INT sink must copy original protocol type to IP header, and strip UDP + INT headers